src/pymax/core.py

```
@@ -252,7 +252,15 @@ async def login_with_code(self, temp_token: str, code: str, start: bool = False)
-        token = resp.get("tokenAttrs", {}).get("LOGIN", {}).get("token")
+
+        login_attrs = resp.get("tokenAttrs", {}).get("LOGIN", {})
+        password_challenge = resp.get("passwordChallenge")
+
+        if password_challenge and not login_attrs:
+            token = await self._two_factor_auth(password_challenge)
+        else:
+            token = login_attrs.get("LOGIN", {}).get("token")
+
         if not token:
             raise ValueError("Login response did not contain tokenAttrs.LOGIN.token")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)